### PR TITLE
Event tracking

### PR DESCRIPTION
**Pull request by ryannealmes**
Some events are now tracked and cause users to be suspended, added and removed from channels in rocketchat.

### CODE REVIEW

File: amd/src/rocketchat.js [hunk 1]
```diff
@@ -25,6 +25,18 @@ define(['core/ajax', 'local_rocketchat/rocketchat'], function(ajax) {
     });
   };
 
+  var set_rocketchat_event_based_sync = function(course_id, eventbasedsync) {
+    var promises = ajax.call([
+      { methodname: 'local_rocketchat_set_rocketchat_event_based_sync', args: { courseid: course_id, eventbasedsync: eventbasedsync } }
```
Comment (atainton) [inline amd/src/rocketchat.js]: can you indent put these on separate lines

File: amd/src/rocketchat.js [hunk 2]
```diff
@@ -25,6 +25,18 @@ define(['core/ajax', 'local_rocketchat/rocketchat'], function(ajax) {
...
+      ]);
+
+    promises[0].done(function(response) {
+      console.log('Save successful');
```
Comment (atainton) [inline amd/src/rocketchat.js]: is this meant to be here?

File: classes/events/observers/user_enrolment_updated.php [hunk 3]
```diff
@@ -2,17 +2,38 @@
 
 namespace local_rocketchat\events\observers;
 
-class user_enrolment_updated {
+class group_member_removed {
```
Comment (atainton) [inline classes/events/observers/user_enrolment_updated.php]: i like what you have done with the separation :)

File: classes/integration/subscriptions.php [hunk 4]
```diff
@@ -10,52 +10,103 @@ class subscriptions
     public $errors = array();
 
     private $client;
+    private $channelapi;
 
     function __construct($client) {
         $this->client = $client;
+        $this->channelapi = new \local_rocketchat\integration\channels($this->client);;
```
Comment (atainton) [inline classes/integration/subscriptions.php]: double ; at the end

File: amd/src/rocketchat.js (hunk 2 above)
Comment (ryannealmes) [inline amd/src/rocketchat.js]: Yes.

File: classes/events/observers/user_enrolment_updated.php (hunk 3 above)
Comment (ryannealmes) [inline classes/events/observers/user_enrolment_updated.php]: The name of this class is wrong though. Going to fix it now.

File: amd/src/rocketchat.js (hunk 1 above)
Comment (ryannealmes) [inline amd/src/rocketchat.js]: I am not really sure what you mean.

Comment (atainton) [inline amd/src/rocketchat.js]: `{ 
methodname: 'local_rocketchat_set_rocketchat_event_based_sync',
 args: { courseid: course_id,
            eventbasedsync: eventbasedsync 
} }`

**Comment by atainton** [comment]
👍

**Pull request merged**